### ISSUE TITLE
Request: Bump Torch Version to 2.7.0 to Fix CUDA Compatibility

### ISSUE THREAD

**Issue by Marcel0024** [TaterTotterson/microWakeWord-Trainer-Nvidia-Docker#4]
I'm currently getting this error: 

<img width="1317" height="454" alt="Image" src="https://github.com/user-attachments/assets/ba9c7bc1-8a79-4bbd-b057-c8bdc1587c73" />

```
The current PyTorch install supports CUDA capabilities sm_50 sm_60 sm_70 sm_75 sm_80 sm_86 sm_90.
If you want to use the NVIDIA GeForce RTX 5060 Ti GPU with PyTorch
```

The issue seems to be in torch and fixed and released in Torch 2.7.0 in april source: https://discuss.pytorch.org/t/my-rtx5080-gpu-cant-work-with-pytorch/217301/4 

The current requirement of this repo is 2.5.1: https://github.com/TaterTotterson/microWakeWord-Trainer-Nvidia-Docker/blob/main/requirements.txt#L2C1-L2C13

Any chance we can have this bumped too atleast 2.7.0?

I'm not familiar with Python, so any other suggestions welcome

**Comment by TaterTotterson** [comment]
just pushed a new build with with cuda 13 and PyTorch 2.9.0 with some other fixes.... haven't had time yet to test

ghcr.io/tatertotterson/microwakeword-trainer-nvidia-docker:testing

dont have a 50 series either so let me know if this works for you

**Comment by TaterTotterson** [comment]
ok I pushed a new updated version with a nice webui

would you mind testing this?

ghcr.io/tatertotterson/microwakeword:latest

new port 8502

**Comment by Marcel0024** [comment]
I saw the tag was updated many times. Any luck?

**Comment by TaterTotterson** [comment]
no the current test build spits put bad models... idk if its because of the newer versions or if I just messed something up 😂
I might play with it more when I have more time

**Comment by 1liminal1** [comment]
I had similar issues with TextyMcSpeechy and fixed it by updating PyTorch to 2.7.1+cu128 inside the container. Not exactly the same issue, but maybe my fix will help:

Why these specific versions matter:
torch 2.7.1+cu128 - First version with RTX 50-series support (sm_120)
pytorch-lightning 1.7.7 - Compatible with PyTorch 2.7.x
torchmetrics 0.11.4 - Works with lightning 1.7.7
numpy 2.0.2 - Max version for numba 0.60.0 compatibility
numba 0.60.0 - First version supporting numpy 2.0+
Without locking these dependencies: Training completes but produces unusable models with poor accuracy and loss values that don't converge properly.

**Comment by AndreKR** [comment]
`ghcr.io/tatertotterson/microwakeword-trainer-nvidia-docker:testing` is no longer available? Running into the sm_120 problem as well.

**Comment by gtjoseph** [comment]
Just tested on 5050...

Started with...
```
sudo podman run --gpus=all -it -p 8888:8888 --network host -v  "/usr/src/home-assistant/vpe/tater/data:/data" \
	--name=tater ghcr.io/tatertotterson/microwakeword:test
```

From the GPU Check cell in the notebook...
```
🔧 Torch CUDA Available: True
  • Device count: 1
  • Current device: 0
  • Device name: NVIDIA GeForce RTX 5050

🔧 ONNX Runtime Providers:
  • ['AzureExecutionProvider', 'CPUExecutionProvider']
⚠️  CUDAExecutionProvider not available — ONNX will fall back to CPU.
```

I still have the container and the data directory.  Any other info I can provide?

**Comment by gtjoseph** [comment]
Gotcha.  I'll let it run all the way through this time.

**Comment by gtjoseph** [comment]
Yay! Training completed (took about 40 minutes on an i7-6950X).  Loaded on a Home Assistant Voice PE and worked fine.

Thanks for this!
